webclient/api/src/main/java/io/helidon/webclient/spi/ClientConnectionCache.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webclient.spi;
+
+import io.helidon.webclient.api.ReleasableResource;
+
+/**
+ * Client connection cache with release shutdown hook to provide graceful shutdown.
+ */
+public abstract class ClientConnectionCache implements ReleasableResource {
+    protected ClientConnectionCache() {
```

Comment:
This will be created also for per-instance caches, so we may leak a lot of shutdown hooks. Please add a hook only for static JVM-wide caches (should be just two - one for HTTP/1 and one for HTTP/2), and do something different for the per-client caches.

Reply:
fixed